backend/web/web_backend/src/cache.rs

```diff
@@ -1,28 +1,197 @@
-use std::cmp::Eq;
 use std::collections::HashMap;
-use std::hash::Hash;
+use std::mem::replace;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
 use tokio::sync::RwLock;
 
-/// TODO: Evaluate if crate `cached` should be used instead.
-/// Put max size on cache.
-/// Put timeout on cache entries.
+struct MaxSize {
+    max_size: usize,
+    filo: Vec<String>,
+    index: usize,
+}
+
+struct Timeout {
+    timeout: Duration,
+    id_to_time: HashMap<String, Instant>,
+}
+
+impl Timeout {
+    fn new(timeout: Duration) -> Self {
+        Timeout {
+            timeout,
+            id_to_time: HashMap::new(),
+        }
+    }
+
+    fn put(&mut self, key: String) {
+        self.id_to_time.insert(key.clone(), Instant::now());
+    }
+
+    fn remove(&mut self, key: String) {
+        self.id_to_time.remove(&key);
+    }
 
-pub struct Cache<K, V> {
-    cache: RwLock<HashMap<K, V>>,
+    // If this is invalid, n
+    fn is_valid(&mut self, key: String) -> bool {
+        if let Some(k) = self.id_to_time.get(&key) {
+            if k.elapsed() >= self.timeout {
+                self.remove(key);
+                return false;
+            }
+        }
+        false
+    }
 }
 
-impl<K: Eq + Hash + Clone, V: Clone> Cache<K, V> {
+// Emluate queue where we can save the Keys from cache
+// As long as the queue is not filled, we save all the data
+
+// When the queue hits maximum, we will start replacing the
+// oldest entries.
+impl MaxSize {
+    fn new(max_size: usize) -> Self {
+        MaxSize {
+            max_size,
+            filo: Vec::new(), // first-in-last-out
```

Comment:
Is the concept of the ring buffer like queue over-engineered?
Would a deque do the work in an easier to read fashion while still maintaining high performance?